sound/soc/sof/loader.c

```diff
@@ -129,6 +130,70 @@ int snd_sof_fw_parse_ext_data(struct snd_sof_dev *sdev, u32 bar, u32 offset)
 }
 EXPORT_SYMBOL(snd_sof_fw_parse_ext_data);
 
+/* parse extended FW manifest data structures */
+static int snd_sof_fw_xman_parse(struct snd_sof_dev *sdev, const void *data)
+{
+	struct xman_header *head = (struct xman_header *)data;
+	struct xman_elem_header *elem_hdr;
+	u32 remaining = head->size - head->header_size;
+	uintptr_t iptr;
+	int ret = 0;
+
+	if (XMAN_VERSION_INCOMPATIBLE(XMAN_VERSION, head->version))
```

Comment:
before checking a version, you should have something that checks if there is an extended manifest in the first place.

Reply:
Added